src/components/utils/PreviewedLink.tsx

```diff
@@ -0,0 +1,146 @@
+    try {
+      const url = new URL(props.href.toString());
+      return url.protocol === 'https:' || url.protocol === 'http:';
+    } catch {
+      return false;
+    }
+  };
+
+  const isTwitter = () => {
+    try {
+      const url = new URL(props.href.toString());
+      return url.protocol === 'https:' && (url.host === 'twitter.com' || url.host === 'x.com');
+    } catch {
+      return false;
+    }
+  };
+
+  const twitterUrl = () => {
+    try {
+      const url = new URL(props.href.toString());
+      url.host = 'twitter.com';
+      return url.href;
+    } catch {
+      return '';
+    }
+  };
+
+  const isYoutube = () => {
+    try {
+      const url = new URL(props.href.toString());
+      return (
+        url.protocol === 'https:' &&
+        url.host === 'www.youtube.com' &&
+        url.pathname === '/watch' &&
+        url.searchParams.get('v')
+      );
+    } catch {
+      return false;
+    }
+  };
+
+  const youtubeUrl = () => {
+    try {
+      const origin = location.href;
+      const url = new URL(props.href.toString());
+      console.log(url.host, url);
+      return 'https://www.youtube.com/embed/' + url.searchParams.get('v') + '?origin=' + origin;
+    } catch {
+      return '';
+    }
+  };
+
+  createEffect(() => {
+    if (isTwitter()) {
+      window.twttr?.widgets.load();
+    }
+  });
+  const initialOgp: OgpContents = { title: '', description: '', image: null, url: null };
+  const [getOgpContents, setOgpContents] = createSignal(initialOgp);
+
+  const fetchOgpContents = async (url: URL) => {
+    const whiteList = ['www3.nhk.or.jp'];
+    if (whiteList.includes(url.host)) {
+      const res = await fetch(url, { headers: { Accept: 'text/html' } });
+      const text = await res.text();
+      const el = new DOMParser().parseFromString(text, 'text/html');
+      const headEls = el.head.children;
+      const ogs = Object.fromEntries(
+        Array.from(headEls)
+          .filter((v) => v.getAttribute('property') && v.getAttribute('content'))
+          .map((v) => [v.getAttribute('property'), v.getAttribute('content')]),
+      );
+      if (ogs['og:image'] && ogs['og:title'] && ogs['og:description']) {
+        const contents: OgpContents = {
+          title: ogs['og:title'],
+          description: ogs['og:description'],
+          image: new URL(ogs['og:image']),
+          url: url,
+        };
+        setOgpContents(contents);
+      }
+    }
+  };
+
+  fetchOgpContents(new URL(props.href.toString()));
+
+  const ogp = getOgpContents();
+  return (
+    <Show when={isSafe()} fallback={props.href}>
+      <Switch fallback={props.href}>
+        <Match when={isTwitter()}>
+          <blockquote class="twitter-tweet">
+            <a class={props.class} href={twitterUrl()} target="_blank" rel="noreferrer noopener">
+              {twitterUrl()}
+            </a>
+          </blockquote>
+        </Match>
+        <Match when={isYoutube()}>
+          <div class="youtube-embed">
```

Comment:
Tailwindを使っているため、以下のように指定してください。

```suggestion
          <div class="w-full aspect-video">
```

- https://tailwindcss.com/docs/width
- https://tailwindcss.com/docs/aspect-ratio